backend/src/server/routes/v1/secret-sharing-router.ts

```diff
@@ -70,6 +70,43 @@ export const registerSecretSharingRouter = async (server: FastifyZodProvider) =>
     }
   });
 
+  server.route({
+    method: "POST",
+    url: "/public",
+    config: {
+      rateLimit: writeLimit
+    },
+    schema: {
+      body: z.object({
+        encryptedValue: z.string(),
+        iv: z.string(),
+        tag: z.string(),
+        hashedHex: z.string(),
+        expiresAt: z
+          .string()
+          .refine((date) => date === undefined || new Date(date) > new Date(), "Expires at should be a future date"),
+        expiresAfterViews: z.number()
+      }),
+      response: {
+        200: z.object({
+          id: z.string().uuid()
+        })
+      }
+    },
+    handler: async (req) => {
+      const { encryptedValue, iv, tag, hashedHex, expiresAt, expiresAfterViews } = req.body;
+      const sharedSecret = await req.server.services.secretSharing.createPublicSharedSecret({
```

Comment:
I've added our existing fastify rateLimiter to this public endpoint with the following shy limits:
```ts
export const publicSecretShareCreationLimit: RateLimitOptions = {
  // Create Shared Secrets
  timeWindow: 60 * 1000,
  max: 5,
  keyGenerator: (req) => req.realIp
};
```

Should I make it more strict? or is there another way to this that you're thinking?

Reply:
I think for this one, we could add a captcha on top of the rate limit? 

If my PR gets merged and we'll indeed use HCaptcha, we could add some sort of an "under the hood" bot tracker, but it requires a PRO plan  
https://github.com/Infisical/infisical/pull/1925

Reply:
Good idea @sheensantoscapadngan. We can do this in a separate PR

Reply:
By the way for this one, it might be risky for us to release this without any other defense mechanism from bots aside from rate limiting. As we've learned, IPs can easily be rotated